src/main/java/com/patchworkmc/mapping/remapper/NaiveRemapper.java

```diff
@@ -4,26 +4,41 @@
 
 import net.fabricmc.tinyremapper.IMappingProvider;
 
+import com.patchworkmc.Patchwork;
+
 /**
- * Remaps classes, methods, and fields based on the assumption that names are never repeated in the source mappings.
+ * Remaps classes, methods, and fields based on the assumption that names are never repeated in the source mappings,
+ * or that they duplicate directly in the target mappings. (i.e. 'valueOf' -> 'valueOf')
+ *
+ * <p>Mappings that are duplicated will be ignored after their first entry.</p>
  */
 public class NaiveRemapper {
 	private final HashMap<String, String> classes = new HashMap<>();
 	private final HashMap<String, String> methods = new HashMap<>();
 	private final HashMap<String, String> fields = new HashMap<>();
+	private final HashMap<String, String> fieldDesc = new HashMap<>();
 
 	public NaiveRemapper(IMappingProvider mappings) {
 		mappings.load(new IMappingProvider.MappingAcceptor() {
 			@Override
 			public void acceptClass(String srcName, String dstName) {
-				if (classes.get(srcName) != null) throw new IllegalArgumentException("Duplicated class name " + srcName);
-				classes.put(srcName, dstName);
+				if (classes.get(srcName) == null) {
+					classes.put(srcName, dstName);
+				} else {
+					throw new IllegalArgumentException("Duplicated class name " + srcName);
+				}
 			}
 
 			@Override
 			public void acceptMethod(IMappingProvider.Member method, String dstName) {
-				if (classes.get(method.name) != null) throw new IllegalArgumentException("Duplicated method name " + method.name);
-				methods.put(method.name, dstName);
+				// Have to include some exceptions
+				String presentName = methods.get(method.name);
+
+				if (presentName == null || presentName.equals(dstName)) {
+					methods.put(method.name, dstName);
+				} else {
+					Patchwork.LOGGER.debug("Duplicated method name %s (proposed %s, but key already mapped to %s!)", method.name, dstName, presentName);
```

Comment:
Why was this downgraded to a log message? Shouldn't we fail fast here?

Reply:
s y n t h e t i c s

Reply:
additionally, Fabric has some intermediary names that Forge leaves as unobf'd (probably ones that become unobf'd in a new release, but were previously)